src/main/java/de/tum/in/www1/artemis/service/exam/StudentExamAccessService.java

```diff
@@ -47,30 +47,26 @@ public StudentExamAccessService(CourseRepository courseRepository, UserRepositor
      * @param courseId      the if of the course
      * @param examId        the id of the exam
      * @param studentExamId the id of the student exam
-     * @param isTestRun     flag to determine if it is a test run or not
      */
-    public void checkStudentExamAccessElseThrow(Long courseId, Long examId, Long studentExamId, boolean isTestRun) {
-        checkStudentExamAccessElseThrow(courseId, examId, studentExamId, userRepository.getUserWithGroupsAndAuthorities(), isTestRun);
+    public void checkStudentExamAccessElseThrow(Long courseId, Long examId, Long studentExamId) {
+        StudentExam studentExam = studentExamRepository.findByIdElseThrow(studentExamId);
+        checkStudentExamAccessElseThrow(courseId, examId, studentExam, userRepository.getUserWithGroupsAndAuthorities());
     }
 
     /**
      * Checks if the current user is allowed to see the requested student exam.
      *
      * @param courseId      the if of the course
      * @param examId        the id of the exam
-     * @param studentExamId the id of the student exam
+     * @param studentExam   the student exam
      * @param currentUser   the current user
-     * @param isTestRun     flag to determine if this is a test run or not
      */
-    public void checkStudentExamAccessElseThrow(Long courseId, Long examId, Long studentExamId, User currentUser, boolean isTestRun) {
-        checkCourseAndExamAccessElseThrow(courseId, examId, currentUser, isTestRun);
-
-        // Check that the student exam exists
-        StudentExam studentExam = studentExamRepository.findByIdElseThrow(studentExamId);
+    public void checkStudentExamAccessElseThrow(Long courseId, Long examId, StudentExam studentExam, User currentUser) {
+        checkCourseAndExamAccessElseThrow(courseId, examId, currentUser, studentExam.isTestRun(), false);
 
         // Check that the examId equals the id of the exam of the student exam
         if (!studentExam.getExam().getId().equals(examId)) {
-            throw new ConflictException("The student exam does not belong to the exam", "StudentExam", "studentExamExamConflict");
+            throw new AccessForbiddenException("The student exam does not belong to the exam");
```

Comment:
we previously agreed to use a ConflictException in such cases

Reply:
hmm.. okay
✅  6855689